internal/controllers/watch/integration_test.go

```diff
@@ -79,6 +79,90 @@ func TestBasics(t *testing.T) {
 	})
 }
 
+func TestIgnoreSideEffects(t *testing.T) {
+	mgr := testutil.NewManager(t)
+	require.NoError(t, NewController(mgr.Manager))
+	mgr.Start(t)
+
+	ctx := testutil.NewContext(t)
+	cli := mgr.GetClient()
+
+	input := &corev1.ConfigMap{}
+	input.Name = "test-input"
+	input.Namespace = "default"
+	require.NoError(t, cli.Create(ctx, input))
+
+	synth := &apiv1.Synthesizer{}
+	synth.Name = "test-synth"
+	synth.Spec.Refs = []apiv1.Ref{{
+		Key: "foo",
+		Resource: apiv1.ResourceRef{
+			Version: "v1",
+			Kind:    "ConfigMap",
+		},
+	}}
+	require.NoError(t, cli.Create(ctx, synth))
+
+	comp := &apiv1.Composition{}
+	comp.Name = "test-comp"
+	comp.Namespace = "default"
+	comp.Spec.Synthesizer.Name = synth.Name
+	comp.Spec.Bindings = []apiv1.Binding{{
+		Key: "foo",
+		Resource: apiv1.ResourceBinding{
+			Name:      input.Name,
+			Namespace: input.Namespace,
+		},
+	}}
+	require.NoError(t, cli.Create(ctx, comp))
+
+	// The initial status is populated
+	var initialResourceVersion string
+	testutil.Eventually(t, func() bool {
+		cli.Get(ctx, client.ObjectKeyFromObject(comp), comp)
+		if len(comp.Status.InputRevisions) != 1 {
+			return false
+		}
+
+		rv := comp.Status.InputRevisions[0].ResourceVersion
+		initialResourceVersion = rv
+		return rv != ""
+	})
+
+	// Start to ignore side effects.
+	comp.Spec.IgnoreSideEffects = true
+	require.NoError(t, cli.Update(ctx, comp))
+
+	// Give some time to the controller to process this.
+	time.Sleep(time.Millisecond * 500)
```

Comment:
wait short time might cause the unit test to be flaky, can we use testuitl.Eventually to check the comp.metadata.resourceVersion is changed after updating the spec to comp?

Reply:
Same as my other comment regarding delays.